napari/components/_dims/dims.py

```diff
@@ -0,0 +1,302 @@
+        """Sets ranges for all dimensions
+
+        Parameters
+        ----------
+        ranges : tuple
+            Ranges of all dimensions
+        """
+        ndim = len(all_ranges)
+        modified_dims = self._ensure_axis_present(ndim-1, no_event=True)
+        self._set_2d_viewing()
+        self.range = all_ranges
+
+        self.events.ndims()
+        for axis_changed in modified_dims:
+            self.events.axis(axis=axis_changed)
+
+    def set_range(self, axis: int, range: Sequence[Union[int, float]]):
+        """Sets the range (min, max, step) for a given axis (dimension)
+
+        Parameters
+        ----------
+        axis : int
+            Dimension index
+        range : tuple
+            Range specified as (min, max, step)
+        """
+        if self.range[axis] != range:
+            self.range[axis] = range
+            self.events.axis(axis=axis)
+
+    def set_point(self, axis: int, value: Union[int, float]):
+        """Sets the point at which to slice this dimension
+
+        Parameters
+        ----------
+        axis : int
+            Dimension index
+        value : int or float
+            Value of the point
+        """
+        if self.point[axis] != value:
+            self.point[axis] = value
+            self.events.axis(axis=axis)
+
+    def set_interval(self, axis: int, interval: Sequence[Union[int, float]]):
+        """Sets the interval used for cropping and projecting this dimension
+
+        Parameters
+        ----------
+        axis : int
+            Dimension index
+        interval : tuple
+            INTERVAL specified with (min, max)
+        """
+        if self.interval[axis] != interval:
+            self.interval[axis] = interval
+            self.events.axis(axis=axis)
+
+    def set_mode(self, axis: int, mode: DimsMode):
+        """Sets the mode: POINT or INTERVAL
+
+        Parameters
+        ----------
+        axis : int
+            Dimension index
+        mode : POINT or INTERVAL
+            Whether dimension is in the POINT or INTERVAL mode
+        """
+        if self.mode[axis] != mode:
+            self.mode[axis] = mode
+            self.events.axis(axis=axis)
+
+    def set_display(self, axis: int, display: bool):
+        """Sets the display boolean flag for a given axis
+
+        Parameters
+        ----------
+        axis : int
+            Dimension index
+        display : bool
+            Bool which is `True` for display and `False` for slice or project.
+        """
+        if self.display[axis] != display:
+            self.display[axis] = display
+            self.events.axis(axis=axis)
+
+    def _set_2d_viewing(self):
+        """Sets the 2d viewing
+        """
+        self.display = [False] * len(self.display)
+        self.display[-1] = True
+        self.display[-2] = True
+
+    def _ensure_axis_present(self, axis: int, no_event=None):
+        """Makes sure that the given axis is in the dimension model
```

Comment:
I'm confused about what this method is doing. It seems like actually updates the model if you give it an axis greater then the max, but the name implies that it just verifies something?

Reply:
I have renamed it to `_add_axes` as I think that is a little more clear - thanks